src/frontend/e2e/pages/filter.ts

```diff
@@ -1,103 +1,96 @@
-import { Page } from "@playwright/test";
-export abstract class FilterSample {
-  public static async applyFilter(
-    page: Page,
-    filterData: Partial<FilterData>
-  ): Promise<void> {
-    // wait for sample page is ready to be handled
-    await page.waitForFunction(() => {
-      const samples = document.querySelector(
-        "a[href$='/data/samples'] > div > div:nth-child(2)"
-      )?.textContent;
-      return parseInt(samples!) > 0;
-    });
-    if (
-      filterData.uploadDateFrom !== undefined ||
-      filterData.uploadDateTo !== undefined
-    ) {
-      await page.locator("button[label='Upload Date']").click();
-      if (filterData.uploadDateFrom !== undefined) {
-        await page
-          .locator("input[name='uploadDateStart']")
-          .fill(filterData.uploadDateFrom);
-      }
-      if (filterData.uploadDateTo !== undefined) {
-        await page
-          .locator("input[name='uploadDateEnd']")
-          .fill(filterData.uploadDateTo);
-      }
-      await page
-        .locator(
-          "//div[not(contains(@style,'visibility: hidden')) and contains(@class,'MuiPaper-root')]/descendant::button[text()='Apply']"
-        )
-        .click();
-    }
-    // select upload date period
-    if (filterData.uploadDatePeriod !== undefined) {
-      await page.locator("button[label='Upload Date']").click();
-      await page
-        .locator("div:not([style*='hidden'])[class*='MuiPaper-elevation'] li")
-        .filter({ hasText: filterData.uploadDatePeriod })
-        .click();
+import { BasePage } from "../pages/basePage";
+
+// until we have reliable data-test-id, we will use
+// these to select date options
+// NOTE: They llok weired but this how PL sees these options
+const applyCollectionDateSelector =
+  "​to​ApplyLast 7 DaysLast 30 DaysLast 3 MonthsLast 6 MonthsLast Year";
+const applyUploadDateSelector = "to​ApplyTodayYesterdayLast 7 Days";
+
+export async function applyFilter(
+  base: BasePage,
+  filterData: Partial<FilterData>
+): Promise<void> {
+  if (
+    filterData.uploadDateFrom !== undefined ||
```

Comment:
instead of all the `!== undefined`, would the TS optional chaining syntax be more succinct? (`filterData?.uploadDateFrom`)

Reply:
Updated